### PR TITLE
Merge Google and OSM GeoPoint activities into one activity using the common MapFragment interface.

### PR DESCRIPTION
**Pull request by zestyping**
#### Overview

This pull request merges `GeoPointMapActivity` (which was Google-specific) and `GeoPointOsmMapActivity` (which is OSM-specific) into a single activity, `GeoPointMapActivity`, using the common `MapFragment` interface added in https://github.com/opendatakit/collect/pull/2465.  The `MapFragment` interface is extended to support point markers, which are used by the activity.

#### Guidance to reviewers

This PR will probably be easiest to review by going through the commits one by one; most of the commits are straightforward.  The biggest change is in [c69bf44](https://github.com/opendatakit/collect/pull/2776/commits/c69bf4413c08dc495bf209cfdaa327f016a5c588), which migrates the new activity to use MapFragment instead of GoogleMap.  Because MapFragment is a higher-level interface, some of the logic and state in the GeoPointMapActivity got simplified away.  If anything in there is confusing, please feel free to reach out to me with any questions—I'm also happy to talk it over with you.  Sometimes a short conversation can really help speed up a review.

#### What has been done to verify that this works as intended?
I've manually tested, in both Google and OSM modes:
  - Zooming to the phone's GPS location
  - Zooming to an existing point
  - Allowing the first GPS fix to determine the point
  - Clearing the point with the "trash" button
  - Placing the point by long-pressing, and confirming that the viewport doesn't move
  - Moving the point by dragging, and confirming that the viewport shifts to center on the new point
  - Placing the point with the "add marker" button, and confirming that the viewport zooms to the point
  - Saving the point
  - Launching with a saved point and confirming that the point cannot be dragged
  - Confirming that dragging and long-pressing is allowed only in "placement-map" mode, not in "map" mode

#### Why is this the best possible solution? Were any other approaches considered?
There is currently a great deal of code duplicated between `GeoPointMapActivity` and `GeoPointOsmMapActivity`, including some vestigial code, member fields that are trivial or unused, and duplicated layouts.  This solution eliminates most of that duplication.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
The only user-visible behaviour changes should be:

  - Tapping on the marker in OSM used to pop up an empty text bubble.  It doesn't do that anymore.
  - When the activity is launched with a previously saved point, marker dragging and long-press are initially disabled; the user must click the "trash" button to discard the point before adding/long-pressing/dragging a new marker.  Previously, the instructions "Long press to place mark to tap add marker button" would be displayed even though the user cannot do these things; now the instructions are not displayed.

#### Do we need any specific form for testing your changes? If so, please attach one.
Any form that uses the GeoPoint widget with "placement-map" or "map" appearance will do.  Here's the one I've been using:
[geo-widgets.zip](https://github.com/zestyping/collect/files/2399351/geo-widgets.zip)

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by zestyping** [comment]
Rebased onto master.

**Comment by zestyping** [comment]
@opendatakit-bot label "needs-testing"

**Comment by zestyping** [comment]
@opendatakit-bot label "needs testing"

**Comment by zestyping** [comment]
The CircleCI failure looks to me like an intermittent network error:
```
Could not GET 'https://jcenter.bintray.com/net/sourceforge/saxon/saxon/9.1.0.8/saxon-9.1.0.8.pom'. Received status code 502 from server: Bad Gateway
```

Is there a way to re-run the checks?

File: collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java [hunk 1]
```diff
@@ -338,37 +371,133 @@ protected void addMapLayoutChangeListener(MapView map) {
...
     }
 
-    protected static @NonNull MapPoint fromGeoPoint(@NonNull GeoPoint geoPoint, double sd) {
-        return new MapPoint(geoPoint.getLatitude(), geoPoint.getLongitude(), geoPoint.getAltitude(), sd);
+    protected static @NonNull MapPoint fromMarker(@NonNull Marker marker) {
+        GeoPoint geoPoint = marker.getPosition();
+        double sd = 0;
+        try {
+            sd = Double.parseDouble(marker.getSubDescription());
+        } catch (NumberFormatException e) { /* ignore */ }
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java]: I think logging at least Timber.w would be good here? We usually don't ignore exceptions.

File: collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java [hunk 2]
```diff
@@ -197,23 +201,45 @@ protected LatLngBounds expandBounds(LatLngBounds bounds, double factor) {
...
+
+    @Override public int addMarker(MapPoint point, boolean draggable) {
+        int featureId = nextFeatureId++;
+        features.put(featureId, new MarkerFeature(map, point, draggable));
+        return featureId;
+    }
+
+    @Override public @Nullable MapPoint getMarkerPoint(int featureId) {
+        MapFeature feature = features.get(featureId);
+        if (feature instanceof MarkerFeature) {
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java]: Don't you like ternary operators? I would prefer it here?

File: collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java [hunk 3]
```diff
@@ -174,28 +181,42 @@ public MapView getMapView() {
...
 
+    @Override public int addMarker(MapPoint point, boolean draggable) {
+        int featureId = nextFeatureId++;
+        features.put(featureId, new MarkerFeature(map, point, draggable));
+        return featureId;
+    }
+
+    @Override public @Nullable MapPoint getMarkerPoint(int featureId) {
+        MapFeature feature = features.get(featureId);
+        if (feature instanceof MarkerFeature) {
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java]: The same like above, ternary operator would be good here.

**Comment by grzesiek2010** [comment]
@zestyping 

I found only some minor things to improve so I'm going to add `needs testing` label. Please let me know once you fix them.

EDIT: since #2797 is based on this branch let's test only that pr.

File: collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java (hunk 1 above)
Comment (zestyping) [reply collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java]: Okay!  I also fixed a similar ignored exception in GoogleMapFragment.

File: collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java (hunk 2 above)
Comment (zestyping) [reply collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java]: Sure!  Sometimes I use them; in this case it didn't occur to me.  Fixed.

**Comment by zestyping** [comment]
Thanks for the review, @grzesiek2010!  I've fixed the things you pointed out.

**Comment by zestyping** [comment]
@kkrawczyk123 @mmarciniak90 Is it helpful to have the old activities still accessible for comparison testing (with the ability to switch between new and old using the volume buttons), or can I remove them so that this PR is in its final state ready for merging?

**Comment by zestyping** [comment]
@grzesiek2010 Hey, I discovered two bugs—so sorry I missed these before your review!

1. If there's an existing point, the activity was only using the latitude and longitude and ignoring the other two fields.  So simply opening and then re-saving a point would lose information.

2. The previous activity's behaviour disables long-press when the activity starts with an existing point.  The new activity didn't match this behaviour.

I've now fixed these in the recent commits.  Could you have another quick look?

**Comment by zestyping** [comment]
Found and fixed two more small issues.  This'll be the last, I hope!

1. The `onStop()` method turns off GPS tracking while the activity is in the background, so there needs to be an `onStart()` method that reactivates the GPS.

2. The `MapHelper helper` field in `GeoPointMapActivity` shadows the `helper` field in the base class, `BaseGeoMapActivity`, and `BaseGeoMapActivity.onSaveInstanceState` crashes when `helper` is null.

**Comment by zestyping** [comment]
This pull request is now obsolete; it has been rolled into #2865.

**Pull request closed without merging**